### PR TITLE
Make sure temp directory is clean during loadAndInstantiate()

### PR DESCRIPTION
**Pull request by lucksus**

### CODE REVIEW

File: bindings/container.cpp [hunk 1]
```diff
@@ -201,6 +201,14 @@ App* Container::loadAndInstantiate(QString path) {
         std::cout << QString("%1 is not a valid Holochain DNA file!").arg(path).toStdString() << std::endl;
         return 0;
     } else {
-        return new App(dna, QStandardPaths::writableLocation(QStandardPaths::TempLocation), this);
+        QString dna_json = QString(holochain_dna_to_json(dna));
+        QCryptographicHash hash(QCryptographicHash::Sha256);
+        hash.addData(dna_json.toUtf8());
```
Comment (Connoropolous) [inline bindings/container.cpp]: What does this line do?

**Comment by Connoropolous** [comment]
Oops. @lucksus I have another idea. We shouldn't merge this into master I don't think. I think we should do #35 first don't you? 

And then merge this into develop instead?

File: bindings/container.cpp (hunk 1 above)
Comment (sphinxc0re) [reply bindings/container.cpp]: This seems to be a hash factory. So you can add data to as much as you like and then call `result()` to actually generate the hash

**Comment by lucksus** [comment]
@Connoropolous, I will have to create a new build for macOS anyway as I made a mistake in the shell script. I would still merge this (and https://github.com/holochain/holosqape/pull/38) the into master and create develop after that... right?

File: bindings/container.cpp (hunk 1 above)
Comment (lucksus) [reply bindings/container.cpp]: exactly

**Comment by Connoropolous** [comment]
That... sorta works. It leaves the dev camp folks a little in the dark, since you're replacing an existing binary without publishing an update, but I think that's a small factor.

**Pull request merged**